### PR TITLE
Mention env templates in nomad init

### PR DESCRIPTION
**Pull request by schmichael**
I kept it simple because the sample is getting so big. Let me know if I should expand it.

### CODE REVIEW

File: command/init.go [hunk 1]
```diff
@@ -337,6 +337,15 @@ job "example" {
       #   change_signal = "SIGHUP"
       # }
 
+      # The "template" stanza can also be used to create env vars for tasks
```
Comment (dadgar) [inline command/init.go]: I would say "environment variables" and also mention the fact that the task will be restarted when the env vars change.

Comment (dadgar) [inline command/init.go]: We should probably make it a validation error to use signal and env at the same time.

**Comment by schmichael** [comment]
Updated

File: nomad/structs/structs.go [hunk 2]
```diff
@@ -3338,6 +3338,9 @@ func (t *Template) Validate() error {
 		if t.ChangeSignal == "" {
 			multierror.Append(&mErr, fmt.Errorf("Must specify signal value when change mode is signal"))
 		}
+		if t.Envvars {
+			multierror.Append(&mErr, fmt.Errorf("cannot use signals with env var templates"))
```
Comment (dadgar) [inline nomad/structs/structs.go]: Test pretty please :)

Comment (schmichael) [inline nomad/structs/structs.go]: eek, sorry. pushed

**Pull request merged**